djautotask/api_rest.py

```diff
@@ -0,0 +1,342 @@
+        if self.integration_code:
+            headers['ApiIntegrationCode'] = self.integration_code
+
+        return headers
+
+    def build_payload(self, **kwargs):
+        filter_array = self._build_filter(**kwargs)
+        query_obj = {'filter': filter_array}
+        self.PAYLOAD = json.dumps(query_obj)
+
+    def _build_filter(self, **kwargs):
+        filter_array = []
+        for condition in kwargs['conditions']:
+            condition_arr = condition.split(",")
+            filter_obj = self._build_filter_obj(*condition_arr)
+            filter_array.append(filter_obj)
+        return filter_array
+
+    def _build_filter_obj(self, field, value, op='eq'):
+        filter_obj = {"op": op, "field": field, "value": value}
+        return filter_obj
+
+    def fetch_resource(self, *args, **kwargs):
+        """
+        Issue a POST request to the specified REST endpoint.
+
+        retry_counter is a dict in the form {'count': 0} that is passed in
+        to verify the number of attempts that were made.
+        """
+        @retry(stop_max_attempt_number=self.request_settings['max_attempts'],
+               wait_exponential_multiplier=RETRY_WAIT_EXPONENTIAL_MULTAPPLIER,
+               wait_exponential_max=RETRY_WAIT_EXPONENTIAL_MAX,
+               retry_on_exception=retry_if_api_error)
+        def _fetch_resource(endpoint, retry_counter=None, **kwargs):
+            if not retry_counter:
+                retry_counter = {'count': 0}
+            retry_counter['count'] += 1
+
+            try:
+                logger.debug('Making POST request to {} with {}'.format(
+                    endpoint, self.PAYLOAD))
+                response = requests.post(
+                    endpoint,
+                    self.PAYLOAD,
+                    timeout=self.timeout,
+                    headers=self.get_headers(),
+                )
+
+            except requests.RequestException as e:
+                logger.error('Request failed: POST {} with {}: {}'.format(
+                    endpoint, self.PAYLOAD, e))
+                raise AutotaskAPIError('{}'.format(e))
+
+            if 200 <= response.status_code < 300:
+                return response.json()
+            elif 400 <= response.status_code < 499:
+                self._log_failed(response)
+                raise AutotaskAPIClientError(
+                    self._prepare_error_response(response))
+            elif response.status_code == 500:
+                self._log_failed(response)
+                raise AutotaskAPIServerError(
+                    self._prepare_error_response(response))
+            else:
+                self._log_failed(response)
+                raise AutotaskAPIError(
+                    self._prepare_error_response(response))
+
+        # PAYLOAD should be the same through the all the requests of any page
+        if not self.PAYLOAD:
+            self.build_payload(**kwargs)
+
+        next_url = args[0]
```

Comment:
As discussed in call, this can be changed to a keyword argument for better readability and maintainability.